### PR TITLE
first draft of patch_wheel.sh to patch a wheel file's metadata

### PR DESCRIPTION
**Pull request by mboisson**
With
./patch_wheel.sh --wheel <while file> --cctag

this adds a "+computecanada" local version to a wheel file's metadata and generates a new wheel. 

This is an utility script intended to address https://github.com/ComputeCanada/software-stack/issues/80

### CODE REVIEW

**Comment by ccoulombe** [comment]
If a wheel name has already a local tag, it should be renamed, not appended.
```
numpy_groupies-0+untagged.325.g2863b82+computecanada-py3-none-any.whl
numpy_groupies-0+untagged.325.g2863b82-py3-none-any.whl
```

**Comment by ccoulombe** [comment]
I suggest using the first part of the name as a prefix for readability in the sed
```
prefix=$(echo $(basename $filename) | cut -d '-' -f1-2 )
version=$(echo $(basename $filename) | cut -d '-' -f2 )
```

File: patch_wheel.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+fullname=$(readlink -f $ARG_WHEEL)
+newname=$fullname
+tmp=$(mktemp --directory)
+pushd $tmp
+unzip -q $fullname
+
+if [[ $ARG_CCTAG -eq 1 ]]; then
+	tag="+computecanada"
+	newname=$(echo $fullname | sed -e "s/\([^-]*\)-\([^-]*\)-\(.*\)\.whl/\1-\2$tag-\3.whl/g")
```
Comment (ccoulombe) [inline patch_wheel.sh:41]: I suggest to use the prefix of the filename.
If a local tag is already present, it should be renamed

**Review by ccoulombe — CHANGES_REQUESTED** [review]

**Comment by ccoulombe** [comment]
But it still works :
```
(31921) wheels_builder (main *) $ pip list
Package        Version                              
-------------- -------------------------------------
numpy          1.20.1+computecanada                 
numpy-groupies 0-untagged.325.g2863b82-computecanada
pip            20.0.2                               
setuptools     46.1.3                               
wheel          0.34.2                               
(31921) wheels_builder (main *) $ pip freeze
numpy==1.20.1+computecanada
numpy-groupies===0-untagged.325.g2863b82-computecanada
```

Some wheels already have the `+computecanada` tag, e.g. `tensorflow_gpu-1.8.0+computecanada-cp35-cp35m-linux_x86_64.whl`

File: patch_wheel.sh (hunk 1 above)
Comment (mboisson) [reply patch_wheel.sh:41]: Why would a local tag be present ?

**Comment by mboisson** [comment]
> But it still works :
> 
> ```
> (31921) wheels_builder (main *) $ pip list
> Package        Version                              
> -------------- -------------------------------------
> numpy          1.20.1+computecanada                 
> numpy-groupies 0-untagged.325.g2863b82-computecanada
> pip            20.0.2                               
> setuptools     46.1.3                               
> wheel          0.34.2                               
> (31921) wheels_builder (main *) $ pip freeze
> numpy==1.20.1+computecanada
> numpy-groupies===0-untagged.325.g2863b82-computecanada
> ```

that `0-untagged.325.g2863b82` is a build tag, not a local version

**Review by ccoulombe — APPROVED** [review]

**Comment by ccoulombe** [comment]
Lgtm 👍

**Pull request merged**